### PR TITLE
Bump euclid and plane-split.

### PR DESCRIPTION
**Pull request by BorisChiou**
We bump euclid for direct-composition, examples, webrender,
webrender_api, and wrench.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/2882)

### CODE REVIEW

**Comment by BorisChiou** [comment]
I notice that only examples, webrender, webrender_api, and wrench are used in Gecko, and webrender_bindings are not in this repo (which is in Gecko, I guess). That's why I didn't update euclid for pathfinder.

**Comment by BorisChiou** [comment]
OK.. taskclaster said I have to update euclid for pathfinder and lyon_geom.

**Comment by nical** [comment]
@BorisChiou ouch I'm sorry you are going through all of this for a non-breaking change. the latest version of lyon_geom already depends on euclid 0.18.0 so if you update pathfinder to that version it should be good (pathfinder being the only crate that depends on lyon_geom). Hopefully the things in lyon_geom that pathfinder use haven't changed too much since the version pathfinder is using. Let me know if you run into issues updating it (I maintain lyon_geom).

File: Cargo.lock [hunk 1]
```diff
@@ -1603,6 +1612,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 "checksum either 1.5.0 (registry+https://github.com/rust-lang/crates.io-index)" = "3be565ca5c557d7f59e7cfcf1844f9e3033650c929c6566f511e8005f205c1d0"
 "checksum env_logger 0.5.10 (registry+https://github.com/rust-lang/crates.io-index)" = "0e6e40ebb0e66918a37b38c7acab4e10d299e0463fe2af5d29b9cc86710cfd2a"
 "checksum euclid 0.17.3 (registry+https://github.com/rust-lang/crates.io-index)" = "c95fd0d455f114291a3109286bd387bd423770058474a2d3f38b712cd661df60"
+"checksum euclid 0.18.2 (registry+https://github.com/rust-lang/crates.io-index)" = "59b34ec7d95d70d5cda27301d6182bc17abce8b5b52e260f5ff32c677923bbb0"
```
Comment (Eijebong) [inline Cargo.lock]: This dupes euclid

**Comment by BorisChiou** [comment]
Saw this error while calling `cargo test` after bump new path-finder and euclid.

error[E0369]: binary operation `*` cannot be applied to type `std::option::Option<euclid::TypedPoint2D<f32, webrender_api::WorldPixel>>`
   --> webrender/src/picture.rs:598:26
    |
598 |     let mut device_pos = world_pos * device_pixel_scale;
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: an implementation of `std::ops::Mul` might be missing for `std::option::Option<euclid::TypedPoint2D<f32, webrender_api::WorldPixel>>

I'm checking this error.

**Comment by BorisChiou** [comment]
OK, let's try again

**Comment by BorisChiou** [comment]
`cargo test` runs successfully on my laptop.

**Comment by Eijebong** [comment]
@BorisChiou This is superseded by #2904 
Sorry, didn't see this one before

**Comment by BorisChiou** [comment]
It's ok. I will close this. Thanks for bumping the version. :)

**Pull request closed without merging**